### DISCUSSION TITLE
2022-02-14 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#1066] [Maintainer discussions]
## About
At the start of each week, we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
- [Last Week in Mathesar](https://github.com/centerofci/mathesar/discussions/1066#discussioncomment-2173079)
- [Google Summer of Code action items](https://github.com/centerofci/mathesar/discussions/1066#discussioncomment-2173081)
- [Frontend priorities](https://github.com/centerofci/mathesar/discussions/1066#discussioncomment-2173085)
- [Backend priorities](https://github.com/centerofci/mathesar/discussions/1066#discussioncomment-2173087)
- [Design priorities](https://github.com/centerofci/mathesar/discussions/1066#discussioncomment-2173089)
- [Product priorities](https://github.com/centerofci/mathesar/discussions/1066#discussioncomment-2173091)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/1066#discussioncomment-2173093)
- [Dominykas' check in](https://github.com/centerofci/mathesar/discussions/1066#discussioncomment-2174477)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/1066#discussioncomment-2173305)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/1066#discussioncomment-2176345)
- [Ghislaine's check in](https://github.com/centerofci/mathesar/discussions/1066#discussioncomment-2176789)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/1066#discussioncomment-2179078)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/1066#discussioncomment-2179664)
- [Mukesh's check in](https://github.com/centerofci/mathesar/discussions/1066#discussioncomment-2180165)

## Weekly check-in
@centerofci/mathesar-core please post your weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

Prompts:
- How did your most recent work go?
- What are you planning to work on soon?
- Do you anticipate any dependencies or blockers that someone else on the team can help you with?

If you'd like to add anything else to your check-in that's not related to the prompts, feel free to do so.

**Comment by kgodey** [comment]
## Last Week in Mathesar

### Feedback Requested
_Nothing at the moment._

### Complete
- **Discussions**:
  - https://github.com/centerofci/mathesar/discussions/1057
  - https://github.com/centerofci/mathesar/discussions/1046
  - https://github.com/centerofci/mathesar/discussions/1035
    - We will renew discussion when we get to #661
  - https://github.com/centerofci/mathesar/discussions/1034
  - https://github.com/centerofci/mathesar/discussions/1007 
    - Feel free to leave comments or thoughts, but we need a draft of #1008 before substantive discussion
- **Wiki updates**:
  - in preparation for our GSoC application, I filled out our [Mentoring](https://wiki.mathesar.org/en/community/mentoring) section, including:
    - [Applicant Guide](https://wiki.mathesar.org/en/community/mentoring/applicant-guide)
    - [Project Proposal Template for applicants](https://wiki.mathesar.org/en/community/mentoring/applicant-guide)
    - [Mentee Guide](https://wiki.mathesar.org/en/community/mentoring/mentee-guide)
    - [Mentor Guide](https://wiki.mathesar.org/en/community/mentoring/mentor-guide)
- **Full stack and general work**:
  - https://github.com/centerofci/mathesar/pull/1041
  - https://github.com/centerofci/mathesar/pull/1058
- **Backend work**:
  - https://github.com/centerofci/mathesar/pull/1022
  - https://github.com/centerofci/mathesar/pull/1060
- **Frontend work**:
  - https://github.com/centerofci/mathesar/pull/1048
  - https://github.com/centerofci/mathesar/pull/1056

**Comment by kgodey** [comment]
## Google Summer of Code action items

I plan to submit our Google Summer of Code organization applications this week. Some action items for everyone:

- Take some time to through our new [Mentoring](https://wiki.mathesar.org/en/community/mentoring) wiki section, especially:
  - [Applicant Guide](https://wiki.mathesar.org/en/community/mentoring/applicant-guide)
  - [Project Proposal Template for applicants](https://wiki.mathesar.org/en/community/mentoring/applicant-guide)
  - [Mentee Guide](https://wiki.mathesar.org/en/community/mentoring/mentee-guide)
  - [Mentor Guide](https://wiki.mathesar.org/en/community/mentoring/mentor-guide)
  - **Action items**:
    - Let me know (here or privately) if any of the documentation looks confusing.
    - Let me know (here or privately) if you have any questions about any of what you read.
- Read through the [draft of our GSoC application](https://hackmd.io/o71R0eNrT8K1rBf4aeW-7A?edit) (note: this is a private document, only accessible to the core team)
  - **Action items**:
    - Let me know (privately) if you have suggested improvements or questions.
- Look through our [Project Ideas](https://wiki.mathesar.org/en/community/mentoring/project-ideas)
  - **Action items**:
    - We need a primary and backup mentor for all ideas. **Post here with the ones you're willing to mentor.**
    - Let me know (here or privately) if anything looks confusing or needs more detail.
- Consider this the last call to add new Google Summer of Code ideas - please add them before end of day Thursday if you have more.

**Reply by dmos62** [reply]
I'm a primary mentor for the automatic hints reflection. I don't mind being a secondary on any of the others; I don't have preferences though.

**Reply by mathemancer** [reply]
My only confusion about the docs is where the entry point is for applicants.  As for as willingness to mentor, I'm happy to mentor or co-mentor on any of the back end tasks, with a preference for data layer things.

**Reply by pavish** [reply]
I'm willing to mentor or co-mentor any project that involves frontend work.

**Reply by silentninja** [reply]
I would love to mentor the following ideas:

- Construct Dependency Graph for Database Objects
- Support More Data Import Sources
- Support Importing Data into Existing Tables

I'm happy to mentor or co-mentor on any other back end related ideas, with a preference for service layer things.

**Reply by kgodey** [reply]
@seancolsen and I talked via DM, I'll put him down as backup mentor or frontend support for any existing ideas that need it.

**Comment by kgodey** [comment]
## Frontend priorities

1. Complete front end changes to https://github.com/centerofci/mathesar/pull/1047
2. Complete #775
3. Complete and merge basic E2E testing: https://github.com/centerofci/mathesar/issues/640. PRs:
	1. https://github.com/centerofci/mathesar/pull/1032
	2. https://github.com/centerofci/mathesar/pull/1062
4. Make progress on issues from the [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) and [2022-02 improvements](https://github.com/centerofci/mathesar/milestone/43) milestones. Some ideas: 
	1. #500
	2. #742
	3. #970
	4. #928
	5. #929
	6. #936
	7. #475

@pavish @seancolsen Please comment below with thoughts/concerns.

**Reply by seancolsen** [reply]
Looks good :grin:

**Reply by pavish** [reply]
Looks good to me

**Comment by kgodey** [comment]
## Backend priorities

The frontend is catching up fast to the backend, so we really need to get backend work for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) completed and start on [Links between Tables](https://github.com/centerofci/mathesar/milestone/47) ASAP. We won't be able to do that this week, but the closer we get, the better.

We met last week and divided up the issues as follows:

1. @dmos62:
    1. **Filtering**:
        1. https://github.com/centerofci/mathesar/issues/385
        2. https://github.com/centerofci/mathesar/issues/406
        3. https://github.com/centerofci/mathesar/issues/410
        4. https://github.com/centerofci/mathesar/issues/413
        5. https://github.com/centerofci/mathesar/issues/418
        6. https://github.com/centerofci/mathesar/issues/427
        7. https://github.com/centerofci/mathesar/issues/431 (currently blocked)
    1. **Data Type API updates:**:
        1. https://github.com/centerofci/mathesar/issues/960
        2. https://github.com/centerofci/mathesar/issues/1036
        3. https://github.com/centerofci/mathesar/issues/1037
2. @mathemancer:
    1. https://github.com/centerofci/mathesar/issues/430
    2. https://github.com/centerofci/mathesar/issues/845 + associated work
    3. **Grouping**:
        1. https://github.com/centerofci/mathesar/issues/408 
        2. https://github.com/centerofci/mathesar/issues/411 
        3. https://github.com/centerofci/mathesar/issues/414
        4. https://github.com/centerofci/mathesar/issues/384
        5. https://github.com/centerofci/mathesar/issues/429
        6. https://github.com/centerofci/mathesar/issues/419
        7. https://github.com/centerofci/mathesar/issues/432 (currently blocked)
3. @silentninja:
    1. **Replace column name usage with ID:**
        1. https://github.com/centerofci/mathesar/issues/839
            1. https://github.com/centerofci/mathesar/issues/992
            2. https://github.com/centerofci/mathesar/issues/993
            3. https://github.com/centerofci/mathesar/issues/994
            4. https://github.com/centerofci/mathesar/issues/995
        2. #896    
    2. https://github.com/centerofci/mathesar/issues/1064
    3. https://github.com/centerofci/mathesar/issues/433 (currently blocked)
    4. Work coming out of https://github.com/centerofci/mathesar/issues/1063 (currently blocked)

I'll also work on #1008 if I have time this week.

@dmos62 @mathemancer @silentninja Please comment below with thoughts/concerns.

**Reply by dmos62** [reply]
Sounds good.

**Reply by mathemancer** [reply]
I think this makes sense

**Comment by kgodey** [comment]
## Design priorities

1. #1063 is top priority since it's blocking backend work.
2. Continue exploratory work for Views in the design prototype.
3. Design specs for:
    1. #1065
    2. #978 

@ghislaineguerin @ppii775 Please comment below with thoughts/concerns

**Reply by ghislaineguerin** [reply]
@kgodey looks good

**Comment by kgodey** [comment]
## Product priorities

1. Finalize definition of Views, which will result in updates to the GitHub issues for Views design and the [product spec for Views](https://wiki.mathesar.org/en/product/specs/2022-01-views).

**Comment by kgodey** [comment]
## Comms Assignee schedule

| Week | Person |
|-|-|
| 2022-02-14 to 2022-02-18 | Dominykas |
| 2022-02-21 to 2022-02-25 | Ghislaine |
| 2022-02-28 to 2022-03-04 | Brent |
| 2022-03-07 to 2022-03-11 | Kriti |
| 2022-03-14 to 2022-03-18 | Pavish |
| 2022-03-21 to 2022-03-25 | Mukesh |
| 2022-03-28 to 2022-04-01 | Sean |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by seancolsen** [comment]
## Sean's check in

How did your most recent work go?

- Going pretty well.
- All the [work](https://github.com/centerofci/mathesar/pull/1032) required to smooth out the E2E testing has been a little frustrating. I'm not really sure how to improve our process though.
- I'm excited to finally have [TypeScript strict mode enabled](https://github.com/centerofci/mathesar/pull/1056).
- I'm still feeling like I ought to be spending a larger portion of my time on building features. Hopefully once the E2E dust is settled that ratio will come into balance.

What are you planning to work on soon?

- Add front end changes to #1047, once the back-end changes are complete.
- #775 (after we have some discussion at this week's [meeting](https://hackmd.io/OEXWY519TceOUi53q0Ay1A))
- #640 (via #1032 and #1062)
- That should probably take me most of the week. If I have time or I get blocked I'll move on to #742.

Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- Nope

Anything else?

- My kids are _finally_ back in school starting this morning!! 🎉

**Reply by kgodey** [reply]
I'm glad the kids are back in school! E2E testing has always been a pain whenever I've worked on it, hopefully the infrastructure you're setting up will make things easier to write new tests.

**Comment by dmos62** [comment]
> How did your most recent work go?

Pretty well. Filtering feels out of the woods.

Ever since we discussed shipping faster, the number of discussions and notifications I have to attend reduced drastically, and it's had a great effect on my focus and productivity. I'm very pleased with the low noise environment I'm enjoying right now. I didn't know I needed it. Also, I'm not sure where the discussions have gone: it's a bit like everyone became more focused. Good stuff!

 > What are you planning to work on soon?

Get foundational filtering PRs merged, then get the type-specific filters added. I'd love to have relevant PRs submitted by end of this week.

 > Do you anticipate any dependencies or blockers that someone else on the team can help you with?

I have two filtering API PRs up for review.

**Reply by kgodey** [reply]
I'm glad to hear that the amount of noise has reduced for you and that it's helpful! :)

**Comment by kgodey** [comment]
> How did your most recent work go?

Last week felt a little slow, I did get more Views work done, but not very much. I switched focus to Google Summer of Code and did get a lot done there, we're almost ready to apply. 

I also have a lot more meetings now with the regular 1:1s and weekly meeting and that is hard on focused product work. I expect it will take me a couple of weeks to adjust fully.

> What are you planning to work on soon?

Other than regular meetings, code review, discussions, etc.:

- Wrapping up the Google Summer of Code application
- Views product work
- If time permits, #1008

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Yes, everyone should prioritize the [Google Summer of Code action items](https://github.com/centerofci/mathesar/discussions/1066#discussioncomment-2173081) since it is time-sensitive.

**Comment by ghislaineguerin** [comment]
> How did your most recent work go?

My most recent work is still in progress. Last week was low on productivity due to taking some days off (I was sick). Had a meeting w/ Kriti at the start of last week that was very helpful in defining priorities for query builder work. 

> What are you planning to work on soon?

- Issue #1063 
- Continue exploratory work for Views in the design prototype.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Comment by pavish** [comment]
## Pavish's check in

> How did your most recent work go?

I was off for most of last week because I was sick. I didn't get any coding done, but did attend meetings, participated in discussions and reviewed some PRs.

> What are you planning to work on soon?

* https://github.com/centerofci/mathesar/issues/254
* https://github.com/centerofci/mathesar/issues/257
* https://github.com/centerofci/mathesar/issues/970
* Add frontend changes to https://github.com/centerofci/mathesar/pull/1069, when ready.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Comment by mathemancer** [comment]
## Brent's check in

> How did your most recent work go?

Last week was less productive due to moving and a sick dog.  However, the `INTERVAL` PR is in.  At some point, we should assess our overall database type infrastructure to see if there are places where we can avoid repeating ourselves.

> What are you planning to work on soon?

I'm speccing out date / time type information passing between the API and UI.  I'll ticket and implement any needed changes.  I'm also trying to squeeze in some of the GSoC tasks.  After that, I'm planning to get into grouping.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Nope.

**Reply by kgodey** [reply]
>  we should assess our overall database type infrastructure to see if there are places where we can avoid repeating ourselves

Please create a ticket to track this. :)

**Comment by silentninja** [comment]
## Mukesh's check in

> How did your most recent work go?

Last week was quite productive, made a PR for `Postgres Money` data type and was able to add additional enhancement to error classes. Weekly check-in was also helpful in resolving a few issues.

> What are you planning to work on soon?

I am planning to complete issues related to replacing column `name` usage with column `id`. Also, help Sean with E2E infrastructure.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Nope.
